torch/distributed/_composable/fsdp/fully_shard.py

```diff
@@ -144,7 +144,20 @@ def reshard(self) -> None:
         Reshards the module's parameters by freeing unsharded parameters if
         needed. This method is *not* recursive.
         """
-        if (state := _get_module_fsdp_state(cast(nn.Module, self))) is None:
-            raise AssertionError(f"No FSDP state found on {self}")
+        state = self._get_fsdp_state()
         if fsdp_param_group := state._fsdp_param_group:
             fsdp_param_group.reshard()
+
+    def set_is_last_backward(self, is_last_backward: bool) -> None:
```

Comment:
We need this API (or similar) anyway for pipeline parallel DFS cooldown backward where there are back-to-back microbatch backwards.

Reply:
is this mostly for PP? for FSDP-only, we would not need this? this is base on my understanding of final post-backward callback of flat-param FSDP

Reply:
We need it any time we are running multiple backwards without a forward. This happens to appear in PP, but for some multi-modal models, I think it could happen with FSDP-only.